Realm/Tests/MigrationTests.mm

```diff
@@ -39,7 +39,7 @@ static void RLMAssertRealmSchemaMatchesTable(id self, RLMRealm *realm) {
         Table *table = objectSchema.table;
         for (RLMProperty *property in objectSchema.properties) {
             XCTAssertEqual(property.column, table->get_column_index(RLMStringDataWithNSString(property.name)));
-            XCTAssertEqual(property.indexed, table->has_search_index(property.column));
+            XCTAssertEqual(property.indexed || property.isPrimary, table->has_search_index(property.column));
```

Comment:
property.indexed indicates if the property was explicitly marked as indexed by the user, and this function was previously not used on things with primary keys.